File: tools/sap/src/main/java/top/bettercode/summer/tools/sap/connection/pojo/EtReturns.java
package top.bettercode.summer.tools.sap.connection.pojo;

import java.util.List;
import org.springframework.util.CollectionUtils;
import top.bettercode.summer.tools.lang.util.StringUtil;
import top.bettercode.summer.tools.sap.annotation.SapTable;

public class EtReturns implements ISapReturn {

  /**
   * 返回消息
   */
  @SapTable("ET_RETURN")
  private List<EtReturn> etReturn;


  /**
   * @return 返回消息
   */
  public List<EtReturn> getEtReturn() {
    return this.etReturn;
  }

  /**
   * 设置返回消息
   *
   * @param etReturn 返回消息
   * @return CostResp
   */
  public EtReturns setEtReturn(List<EtReturn> etReturn) {
    this.etReturn = etReturn;
    return this;
  }

  @Override
  public String toString() {
    return StringUtil.json(this);
  }

  //--------------------------------------------

  @Override
  public boolean isSuccess() {
    if (CollectionUtils.isEmpty(etReturn)) {
      return false;
    } else {
      return "S".equals(etReturn.get(0).getType());
    }
  }


  @Override
  public String getMessage() {
    if (CollectionUtils.isEmpty(etReturn)) {
      return null;
    } else {
      return etReturn.get(0).getMessage();
    }
  }
}

File: tools/sap/src/main/kotlin/top/bettercode/summer/tools/sap/connection/pojo/EtReturns.kt
package top.bettercode.summer.tools.sap.connection.pojo

import org.springframework.util.CollectionUtils
import top.bettercode.summer.tools.lang.util.StringUtil.json
import top.bettercode.summer.tools.sap.annotation.SapTable

open class EtReturns : ISapReturn {
    /**
     * @return 返回消息
     */
    /**
     * 返回消息
     */
    @SapTable("ET_RETURN")
    var etReturn: List<EtReturn>? = null
        private set

    /**
     * 设置返回消息
     *
     * @param etReturn 返回消息
     * @return CostResp
     */
    fun setEtReturn(etReturn: List<EtReturn>?): EtReturns {
        this.etReturn = etReturn
        return this
    }

    override fun toString(): String {
        return json(this)
    }

    override val isSuccess: Boolean
        //--------------------------------------------
        get() = if (CollectionUtils.isEmpty(etReturn)) {
            false
        } else {
            "S" == etReturn!![0].type
        }
    override val message: String?
        get() = if (CollectionUtils.isEmpty(etReturn)) {
            null
        } else {
            etReturn!![0].type
        }
}